Convective boundary

simonguichandut opened the pull request.

jorafb commented:
this model does not compile anymore (it used to in previous versions of MaestroEX)

zingale commented:
I can get it to work with #488

simonguichandut commented:
It compiles on my end so it might be a compiler version issue

jorafb commented:
Hi Mike,

The code compiles great on NASA Pleiades, but the initial conditions (background state) from plt_InitData does not match what it is given analytically to the code.

simonguichandut commented:
I took a quick look and updated a few things, like how to specify the eos parameters in the inputs file. 

I also added print statements in `MaestroBaseState.cpp` and validated that we are putting in the intended model. The EOS also computes the right isothermal temperature every where with $T=T(\rho,p,X)$.  Well not quite, it finds 163.98 K instead of 165 K. I don't think that's a big issue for now..

What Rafa is referring to is that when we load the first model (`plt_InitData`) into `yt`, we get completely different results. The density seems to have the right profile but the order of magnitude is off (in fact all the variables extracted by yt seem to be dimensionless).  The temperature has a huge gradient and the pressure is constant, the exact opposite of what is expected!

It's worth noting that this thing used to work back when Alice first put it together, so we must not be adapting correctly to changes in either Maestroex or Microphysics. @zingale any ideas?
